### PR TITLE
No longer sending events for frames you pass through on the way to other frames.

### LINKED ISSUE #12: `onFrameChange()` is very inefficient in combination with `goToFrame(0)`
In `src/demo/demo.ts` add `console.log({ frameIndex })` to the `onFrameChange` function.

code sample

Open the browser console and jump forward a few frames.

Now press the "start" button and watch what gets logged to the console.

Clicking start triggers on frame change for every frame it passes

Note that `onFrameChange` gets called for every single frame it passes on it's way to it's destination. This is highly inefficient and not expected behavior.

`onFrameChange` should only get called once it has reached its final destination.

I think this issue might be what is causing #9 for me.

### PR DESCRIPTION
**Pull request by iangilman**
Fixes https://github.com/pixfabrik/driftory-viewer/issues/12

### CODE REVIEW

**Comment by iangilman** [comment]
...And published!

**Comment by iangilman** [comment]
Hmm... actually the publish failed: 
```
[13:43:23] Starting 'js:npm-bundle'...
/Users/iangilman/projects/comics/driftory-viewer/node_modules/@dan503/load-js/index.ts(26,2): error TS2322: Type '{ hasLoaded: boolean; callbacks: Function[]; } | {}' is not assignable to type '{ hasLoaded: boolean; callbacks: Function[]; }'.
  Type '{}' is missing the following properties from type '{ hasLoaded: boolean; callbacks: Function[]; }': hasLoaded, callbacks
TypeScript: 1 semantic error
TypeScript: emit succeeded (with errors)
[13:43:25] 'js:npm-bundle' errored after 2.07 s
[13:43:25] Error: TypeScript: Compilation failed
    at Output.mightFinish (/Users/iangilman/projects/comics/driftory-viewer/node_modules/gulp-typescript/release/output.js:131:43)
    at /Users/iangilman/projects/comics/driftory-viewer/node_modules/gulp-typescript/release/output.js:66:22
[13:43:25] 'npm-bundle' errored after 2.07 s
```

@Dan503 Any thoughts on how to fix this? Looks like the problem is in your library?

File: src/demo/demo.ts [hunk 1]
```diff
@@ -1,13 +1,19 @@
 import Driftory from '../library/driftory';
 
 document.addEventListener('DOMContentLoaded', () => {
-  const container = document.querySelector('.driftory-viewer-container') as HTMLDivElement;
+  const container = document.querySelector('.driftory-viewer-container') as HTMLDivElement | null;
```
Comment (Dan503) [inline src/demo/demo.ts:4]: Just remove the `as HTMLDivElement | null`

If you are going to check for existence then this isn't really necessary.

**Comment by Dan503** [comment]
@iangilman try running npm install. You might have an old version of load-js on your computer maybe.

Otherwise I'll need to look into it.

**Comment by Dan503** [comment]
@iangilman actually yeah, it's a bug in my library. 

https://github.com/Dan503/load-js/blob/master/index.ts

Lines 5 & 6 should list those as optional properties.

**Comment by Dan503** [comment]
Fixed in PR #15

File: src/demo/demo.ts (hunk 1 above)
Comment (Dan503) [reply src/demo/demo.ts:4]: OK, I tried making this change but `document.querySelector` returns an `Element` type, not a `HTMLElement` type and the stye prop doesn't exist on the `Element` type so we have no choice but to do this.

Comment (iangilman) [reply src/demo/demo.ts:4]: Yup, that's the same process I went through :-) thank you for confirming!

Comment (Dan503) [reply src/demo/demo.ts:4]: We should probably add a comment there to stop people going through the same loop we both just went through 😕

Comment (iangilman) [reply src/demo/demo.ts:4]: Okay, I've added a comment as part of https://github.com/pixfabrik/driftory-viewer/pull/20

**Pull request merged**